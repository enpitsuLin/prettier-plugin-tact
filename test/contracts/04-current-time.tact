import "@stdlib/deploy";

contract CurrentTime with Deployable {

    deployTime: Int as uint32;

    init() {
        self.deployTime = now(); // returns unix time, the number of seconds since the epoch
    }

    receive("wait 10s") {
        require(now() - self.deployTime > 10, "Did not wait long enough");
        dump("thanks for waiting 10 seconds");
    }

    receive("wait 10d") {
        require(now() - self.deployTime > 10*24*60*60, "Did not wait long enough");
        dump("thanks for waiting 10 days");
    }

    get fun unixTime(): Int {
        return now();
    }

    get fun stringTime(): String {
        let sb: StringBuilder = beginString();
        sb.append(now().toString());
        sb.append(" seconds elapsed since 1 January 1970");
        return sb.toString();
    }
}