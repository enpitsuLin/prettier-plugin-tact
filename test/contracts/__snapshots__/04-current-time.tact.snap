import "@stdlib/deploy";

contract CurrentTime with Deployable {
    deployTime: Int as uint32;

    receive("wait 10s") {
        require(now() - self.deployTime > 10, "Did not wait long enough");dump("thanks for waiting 10 seconds");
    }

    receive("wait 10d") {
        require(now() - self.deployTime > 10*24*60*60, "Did not wait long enough");dump("thanks for waiting 10 days");
    }

    get fun unixTime(): Int {
        return ;
    }

    get fun stringTime(): String {
        let sb: StringBuilder = ;
        sb.append(now().toString());sb.append(" seconds elapsed since 1 January 1970");return sb.toString();
    }
}